sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/UnwrapCastInBinaryComparison.scala

```diff
@@ -155,7 +155,16 @@ object UnwrapCastInBinaryComparison extends Rule[LogicalPlan] {
       list.foreach {
         case lit @ Literal(null, _) => nullList += lit
         case lit @ NonNullLiteral(_, _) =>
-          unwrapCast(EqualTo(in.value, lit)) match {
```

Comment:
shall we make `unwrapCast` return Option so that the code can be clearer?

Reply:
moved to branch `case equalTo @ EqualTo(_, unwrapLit: Literal)`.

Reply:
I mean something like
```
val newList = list.map(lit => unwrapCast(EqualTo(in.value, lit)))
if (newList.forall(_.isDefined)) {
  ...
} else {
  exp
}
```

Reply:
since `unwrapCast` always returns a non-null value, it seems that these changes don't make the code clearer?